src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/ExpressionCompilerTests.cs

```diff
@@ -6309,5 +6309,78 @@ .maxstack  1
 }");
         });
         }
+
+        [Fact]
+        public void RefReadOnlyLambdasEvaluationWillSynthesizeRequiredAttributes_Parameters()
+        {
+            var reference = CreateStandardCompilation(@"
+public delegate void D(ref readonly int p);");
+
+            CompileAndVerify(reference, symbolValidator: module =>
+            {
+                Assert.NotNull(module.ContainingAssembly.GetTypeByMetadataName(AttributeDescription.CodeAnalysisEmbeddedAttribute.FullName));
+                Assert.NotNull(module.ContainingAssembly.GetTypeByMetadataName(AttributeDescription.IsReadOnlyAttribute.FullName));
+            });
```

Comment:
Consider extracting the `module => { ... }` lambda to a helper method.

Reply:
They're doing the opposite. First we assert it is there, and then assert it is not there.